source/backend/api/Services/AcquisitionFileService.cs

```diff
@@ -281,6 +289,7 @@ public PimsCompensationRequisition AddCompensationRequisition(long acquisitionFi
             _logger.LogInformation("Adding compensation requisition for acquisition file id ...", acquisitionFileId);
 
             _user.ThrowIfNotAuthorized(Permissions.CompensationRequisitionAdd);
+            _user.ThrowInvalidAccessToAcquisitionFile(_userRepository, _acqFileRepository, acquisitionFileId);
```

Comment:
I think we need this on GetAcquisitionCompensations, UpdateAgreements, GetAgreement, Add, GetChecklistItems, GetOwners, GetProperties

Reply:
@eddherrera with this change I think we are good.